lil_bank/accounts/forms.py

```diff
@@ -19,13 +21,39 @@ class SignUpForm(forms.Form):
     """
     first_name = forms.CharField(max_length=32)
     last_name = forms.CharField(max_length=32)
+    last_name = forms.CharField(max_length=32)
     username = forms.CharField(max_length=50)
     email = forms.EmailField(max_length=254)
     address = forms.CharField(max_length=256)
     phone = forms.CharField(max_length=16)
     password1 = forms.CharField(label="Password", strip=False, widget=forms.PasswordInput,min_length=8)
     password2 = forms.CharField(label="Re-enter Password", strip=False, widget=forms.PasswordInput,min_length=8)
 
+    def clean_first_name(self):
+        first_name = self.cleaned_data['first_name']
+        if not re.match(r'^[a-zA-Z]+$', first_name):
+            raise forms.ValidationError("First name must contain only letters.")
+        return first_name
+    
+    def clean_last_name(self):
+        last_name = self.cleaned_data['last_name']
+        if not re.match(r'^[a-zA-Z]+$', last_name):
+            raise forms.ValidationError("Last name must contain only letters.")
+        return last_name
+    
+    def clean_phone(self):
+        phone = self.cleaned_data['phone']
+        if not re.match(r'^[0-9]+$', phone):
```

Comment:
While this method of validating the phone number works to a certain degree, there are two approaches that we can take to improve upon it.
1. I think it's best to restrict the phone number to 10 digits. That said, the RegEx for the same would be `^\d{10}$`.
2. If we plan on considering the cases of country codes and the such, then there will be a need for another field for the same. Keeping them separate allows for extensibility in the future to segregate phone numbers on a country basis.

However, implementing the second point might be redundant if it's a country wide deployment. Changes required,
- [ ] RegEx to fit 10 digits only.

Reply:
Done